src/Application/Actions/ActionError.php

```diff
@@ -18,22 +19,15 @@ class ActionError implements JsonSerializable
     public const VALIDATION_ERROR = 'VALIDATION_ERROR';
     public const VERIFICATION_ERROR = 'VERIFICATION_ERROR';
 
-    private string $type;
-    private ?string $description;
-
-    /**
-     * @param string        $type
-     * @param string|null   $description
-     */
-    public function __construct(string $type, ?string $description)
-    {
-        $this->type = $type;
-        $this->description = $description;
-    }
+    public function __construct(
+        public string $type,
+        public ?string $description)
+    {}
```

Comment:
@NoelLH Just so I know i'm on the right track, does something like this make sense r.e `Constructor property promotion` ?

Reply:
@tbg-jasonfung the access modifiers should be the same as on the property declaration you're replacing, but I think otherwise yep